nj2k/src/org/jetbrains/kotlin/nj2k/conversions/ContinueStatementConversion.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.nj2k.conversions
+
+import com.intellij.psi.*
+import org.jetbrains.kotlin.nj2k.*
+import org.jetbrains.kotlin.nj2k.tree.*
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+// This conversion is for converting continue statements. For example, this adds updaters in converted while loops, and attaches
+// labels named 'loop' to continue statements inner 'when' in loops and their corresponding loops for avoiding syntax errors.
+// If the 'continue' keywords in 'switch' is allowed by the kotlin syntax, only converting
+// JKKtConvertedFromForLoopSyntheticWhileStatement will be enough.
+class ContinueStatementConversion(context: NewJ2kConverterContext) : RecursiveApplicableConversionBase(context) {
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element is JKLoopStatement) {
+            convert(element)?.also { return recurse(it) }
+        }
+
+        return recurse(element)
+    }
+
+    private fun convert(loopStatement: JKLoopStatement): JKStatement? {
+        // Check JKKtConvertedFromForLoopSyntheticWhileStatement and not JKWhileStatement if its parent is
+        // JKKtConvertedFromForLoopSyntheticWhileStatement because setting LabeledExpression to the child of
+        // JKKtConvertedFromForLoopSyntheticWhileStatement causes type error.
+        if (loopStatement is JKWhileStatement && loopStatement.parent is JKKtConvertedFromForLoopSyntheticWhileStatement) {
+            return null
+        }
+
+        var needLabel = false
+        var labelChanged = false
+        var loopLabel = "loop"
```

Comment:
So we really need so many mutable states?

Reply:
I thought that if the label is always single one, it is ok with flipping one boolean state, however, multiple states are needed if some labels are there at the same time.
But I'm now not confident about that...